src/ui/public/parse_query/lib/to_user.js

```diff
@@ -10,6 +10,7 @@ export default function toUser(text) {
   if (text == null) return '';
   if (_.isObject(text)) {
     if (text.query_string) return toUser(text.query_string.query);
+    if (text.match_all && !_.has(text.match_all, 'boost')) return '';
     return angular.toJson(text);
   }
   if (text === '*') {
```

Comment:
(I'm referring to [these lines](https://github.com/trevan/kibana/blob/1922565f6cf0117973168131bc661c8755cb9d97/src/ui/public/parse_query/lib/to_user.js#L16-L18).)